packages/state-transition/src/util/execution.ts

```diff
@@ -85,43 +96,33 @@ export function isExecutionCachedStateType(state: CachedBeaconStateAllForks): st
   return (state as CachedBeaconStateExecutions).latestExecutionPayloadHeader !== undefined;
 }
 
-/** Type guard for allForks.ExecutionBlockBody */
-export function isExecutionBlockBodyType(
-  blockBody: allForks.BeaconBlockBody
-): blockBody is allForks.ExecutionBlockBody {
-  return (blockBody as allForks.ExecutionBlockBody).executionPayload !== undefined;
+/** Type guard for ExecutionBlockBody */
+export function isExecutionBlockBodyType(blockBody: BeaconBlockBody): blockBody is BeaconBlockBody<ForkExecution> {
+  return (blockBody as BeaconBlockBody<ForkExecution>).executionPayload !== undefined;
 }
 
-export function getFullOrBlindedPayload(
-  block: allForks.FullOrBlindedBeaconBlock
-): allForks.FullOrBlindedExecutionPayload {
+export function getFullOrBlindedPayload(block: BeaconBlock): ExecutionPayload<ForkExecution, FullOrBlinded> {
   return getFullOrBlindedPayloadFromBody(block.body);
 }
 
 export function getFullOrBlindedPayloadFromBody(
-  body: allForks.FullOrBlindedBeaconBlockBody
-): allForks.FullOrBlindedExecutionPayload {
+  body: BeaconBlockBody<ForkAll, FullOrBlinded>
```

Comment:
can we use `BeaconBlockBody<ForkExecution>` here?

Reply:
At the context of this function we don't know the parameter belongs to which fork, so why I tend to use `ForkAll`